src/test/java/com/example/demo/model/CartTest.java

```diff
@@ -0,0 +1,101 @@
+package com.example.demo.model;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class CartTest {
+
+    private Product product1;
+    private Product product2;
+    private ProductOption productOption1;
+    private ProductOption productOption2;
+
+    @BeforeEach
+    void setUp() {
+        product1 = new Product(new ProductId("product-1"), "product #1", 5000);
+        product2 = new Product(new ProductId("product-2"), "product #2", 3000);
+
+        productOption1 = new ProductOption("red", "M");
+        productOption2 = new ProductOption("black", "L");
+    }
+
+    @Test
+    @DisplayName("빈 장바구니의 수량은 0이다.")
+    void cartTotalQuantityIsZero() {
+        Cart cart = new Cart(List.of());
+
+        assertThat(cart.getTotalQuantity()).isEqualTo(0);
+    }
+
+    @Test
+    @DisplayName("빈 장바구니에 물건 추가하면 장바구니의 전체수량은 추가한 수량과 같다.")
+    void addProduct() {
+        Cart cart = new Cart(List.of());
+
+        int quantity = 1;
+        cart.addProduct(product1.getId(), productOption1, quantity);
+
+        assertThat(cart.getTotalQuantity()).isEqualTo(quantity);
+    }
+
+    @Test
+    @DisplayName("장바구니에 이미 있는 물건 추가하면 전체 수량은 이미 있던 수량과 새로 추가하는 수량의 합이다.")
+    void addExistingProduct() {
+
+        int oldQuantity = 1;
+        Cart cart = new Cart(List.of(
+                createLineItem(product1, productOption1, oldQuantity)
+        ));
+
+        int newQuantity = 1;
+        cart.addProduct(product1.getId(), productOption1, newQuantity);
```

Comment:
`new Cart(어쩌고)`는 테스트 코드를 읽기 너무 어렵습니다. 실제 제품 코드에서 `new Cart(어쩌고)`를 쓸 일도 없고요.

다음과 같이 하는 게 자연스럽게 읽힙니다.

```suggestion
        int oldQuantity = 1;
        int newQuantity = 1;

        Cart cart = new Cart();

        cart.addProduct(product1.getId(), productOption1, oldQuantity);

        cart.addProduct(product1.getId(), productOption1, newQuantity);
```

이건 아래에 있는 모든 다른 테스트 코드에도 마찬가지입니다. 우리는 두 개의 addProduct 메서드 호출하는 부분의 차이점만 순식간에 읽으면 됩니다. 지금처럼 new Cart에 뭐가 들어갔는데 그 다음에 addProduct는 또 어떻게 됐는지 굉장히 다른 모양으로 있는 것 사이의 차이점을 찾으려고 노력하지 않아도 너무 쉽게 알 수 있게 되죠.